embassy-stm32/src/gpio.rs

```diff
@@ -55,6 +55,14 @@ impl From<Speed> for vals::Ospeedr {
     }
 }
 
+/// Type settings
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum OutputType {
```

Comment:
This is only used in the private `set_as_af` API, so it should be `pub(crate)`

Reply:
If I do this I get:
```
crate-private type `gpio::OutputType` in public interface
```

Reply:
oh, it's because `sealed::Pin` is "pub but not really" due to `pub(crate) mod sealed`. 

Moving it inside `mod sealed` should work.

Reply:
`set_as_af` is public, is it not?

Reply:
If I move it inside `sealed::Pin` it's not seen in all the other external modules: `i2c`, `spi`, `uart`, `bxcan`, etc

Reply:
`Pin` is a [sealed trait](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed), to forbid external crates from implementing it.

It's split in two: a private `trait gpio::sealed::Pin`, and a public `trait gpio::Pin: gpio::sealed::Pin`. 

Both are `pub`, but `mod sealed` is `pub(crate)`. This makes it so `sealed::Pin` is inaccessible from outside the crate even if it's `pub`. This is what prevents outside crates from implementing it.

This also makes it so that methods in `sealed::Pin` are effectively crate-private, they're not accessible from outside the crate either.